public/app/plugins/datasource/grafana-azure-monitor-datasource/credentials.ts

```diff
@@ -0,0 +1,154 @@
+import { AzureCredentials, AzureDataSourceInstanceSettings, AzureDataSourceSettings } from './types';
+
+export function getAzureCloud(options: AzureDataSourceSettings | AzureDataSourceInstanceSettings): string {
+  return options.jsonData.cloudName || 'azuremonitor';
+}
+
+function getSecret(options: AzureDataSourceSettings): undefined | string | object {
+  if (options.secureJsonFields.clientSecret) {
+    // The secret is concealed on server
+    return {};
```

Comment:
Is the object return value used as like a marker that the value is concealed?

Maybe an enum of a Symbol might be a bit clearer here?